portality/tasks/preservation.py

```diff
@@ -0,0 +1,589 @@
+import csv
+import hashlib
+import json
+import logging
+import os
+import requests
+import shutil
+import tarfile
+from bagit import make_bag, BagError
+from copy import deepcopy
+from datetime import datetime
+from zipfile import ZipFile
+
+from portality.background import BackgroundTask, BackgroundApi
+from portality.core import app
+from portality.decorators import write_required
+from portality.lib import dates
+from portality.models import Article, BackgroundJob, PreservationState
+from portality.regex import DOI_COMPILED, HTTP_URL_COMPILED
+from portality.tasks.redis_huey import main_queue, configure
+
+
+class PreservationException(Exception):
+    """~~PreservationException:Exception~~"""
+    pass
+
+
+class PreservationStorageException(Exception):
+    pass
+
+
+class ValidationError(Exception):
+    pass
+
+
+class PreservationBackgroundTask(BackgroundTask):
+    """~~PreservationBackground:Feature~~"""
+
+    __action__ = "preserve"
+
+    @classmethod
+    def prepare(cls, username, **kwargs):
+        """
+        ~~->Prepare:Feature~~
```

Comment:
I think this feature annotation is a bit too general, I think we can just leave it out